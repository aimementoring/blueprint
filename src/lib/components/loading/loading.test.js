import React from 'react';
import renderer from 'react-test-renderer';
import Loading from './loading';

describe('Loading false', () => {
  it('renders properly', () => {
    const tree = renderer.create(<Loading loading={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Loading true', () => {
  it('renders properly', () => {
    const tree = renderer.create(<Loading loading={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
