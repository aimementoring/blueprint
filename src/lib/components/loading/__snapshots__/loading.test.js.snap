// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading false renders properly 1`] = `null`;

exports[`Loading true renders properly 1`] = `
<div
  className="theme-base"
>
  <div
    className="loading"
  >
    <svg
      className="loadingSvg"
      viewBox="0 0 1000 1000"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs>
        <linearGradient
          id="gradient"
          x1="0%"
          x2="0%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            stopColor="#9135F0"
          />
          <stop
            offset="100%"
            stopColor="#5eaefd"
          />
        </linearGradient>
      </defs>
      <title>
        … Loading …
      </title>
      <g>
        <path
          className="loader letterA"
          d="M76.92,469.91C76.92,219,280.35,15.53,531.3,15.53S985.68,219,985.68,469.91,782.25,924.29,531.3,924.29C365.73,924.29,132,773.82,178,626.51c0,0,96.6-318.12,120.56-329.3,33.49-15.62-.47,121.68,35.25,300.28"
        />
        <path
          className="loader letterADash"
          d="M192.51,507.07s151.26-84.88,183.63-117.25"
        />
        <path
          className="loader letterI"
          d="M435.86,334.05s-38,214.33-36.84,251.16"
        />
        <path
          className="loader letterM"
          d="M456,590.79c20.1-27.9,52.56-219.9,73.72-218.79s26.75,168.56,42.38,167.45S668.6,296,694.28,296,648,602.51,659.12,643.82"
        />
        <path
          className="loader letterE"
          d="M884.6,359.17s-79.81,29-107.72,46.32c-15.14,9.39-33.24,54.28-44.23,95.79-9.25,35-13.45,67.55-6.56,74.44,15.07,15.07,42.42,36.84,147.35-27.9"
        />
        <path
          className="loader letterEDash"
          d="M732.65,501.28C764.05,508.19,850,480.84,850,480.84"
        />
        <path
          className="loader letterAA"
          d="M76.92,469.91C76.92,219,280.35,15.53,531.3,15.53S985.68,219,985.68,469.91,782.25,924.29,531.3,924.29C365.73,924.29,132,773.82,178,626.51c0,0,96.6-318.12,120.56-329.3,33.49-15.62-.47,121.68,35.25,300.28"
        />
        <path
          className="loader letterAADash"
          d="M192.51,507.07s151.26-84.88,183.63-117.25"
        />
        <path
          className="loader letterII"
          d="M435.86,334.05s-38,214.33-36.84,251.16"
        />
        <path
          className="loader letterMM"
          d="M456,590.79c20.1-27.9,52.56-219.9,73.72-218.79s26.75,168.56,42.38,167.45S668.6,296,694.28,296,648,602.51,659.12,643.82"
        />
        <path
          className="loader letterEE"
          d="M884.6,359.17s-79.81,29-107.72,46.32c-15.14,9.39-33.24,54.28-44.23,95.79-9.25,35-13.45,67.55-6.56,74.44,15.07,15.07,42.42,36.84,147.35-27.9"
        />
        <path
          className="loader letterEEDash"
          d="M732.65,501.28C764.05,508.19,850,480.84,850,480.84"
        />
      </g>
    </svg>
  </div>
</div>
`;
